packages/wrangler/src/hyperdrive/client.ts

```diff
@@ -9,20 +9,46 @@ export type HyperdriveConfig = {
 	caching: CachingOptions;
 };
 
-export type PublicOrigin = {
-	host?: string;
-	port?: number;
-	scheme?: string;
-	database?: string;
-	user?: string;
-	access_client_id?: string;
+export type OriginCommon = {
+	host: string;
+	scheme: string;
+	database: string;
+	user: string;
 };
 
-export type OriginWithSecrets = PublicOrigin & {
-	password?: string;
-	access_client_secret?: string;
+export type OriginCommonWithSecrets = OriginCommon & {
+	password: string;
 };
 
+export type OriginHoA = OriginCommon & {
+	access_client_id: string;
+	access_client_secret?: never;
+	port?: never;
+};
+
+export type OriginHoAWithSecrets = OriginCommonWithSecrets & {
+	access_client_id: string;
+	access_client_secret: string;
+	port?: never;
+};
+
+export type OriginHostAndPort = OriginCommon & {
```

Comment:
In SQC-Config, the "HostAndPort" naming convention is what we use for non-HoA configs. Having the HoA-specific fields here seems wrong?

Reply:
This is doing the same thing as OriginHoOA, instead enforcing that the HoA fields are not present